st2client/st2client/commands/keyvalue.py

```diff
@@ -269,37 +269,61 @@ def __init__(self, resource, *args, **kwargs):
                                                       help_text, *args, **kwargs)
 
         self.parser.add_argument(
-            'file', help=('JSON file containing the %s to create.'
+            'file', help=('JSON/YAML file containing the %s(s) to load'
                           % resource.get_plural_display_name().lower()))
 
     @resource.add_auth_token_to_kwargs_from_cli
     def run(self, args, **kwargs):
+        # normalize the file path to allow for relative files to be specified
         file_path = os.path.normpath(pjoin(os.getcwd(), args.file))
 
-        if not os.path.exists(args.file):
-            raise ValueError('File "%s" doesn\'t exist' % (file_path))
+        # load the data (JSON/YAML) from the file
+        kvps = resource.load_meta_file(file_path)
 
-        if not os.path.isfile(args.file):
-            raise ValueError('"%s" is not a file' % (file_path))
-
-        with open(file_path, 'r') as f:
-            kvps = json.loads(f.read())
+        # if the data is not a list (ie. it's a single entry)
+        # then make it a list so our process loop is generic
+        if not isinstance(kvps, list):
+            kvps = [kvps]
 
         instances = []
         for item in kvps:
+            # parse required KeyValuePair properties
             name = item['name']
             value = item['value']
 
+            # parse optional KeyValuePair properties
+            scope = item.get('scope', DEFAULT_SCOPE)
+            user = item.get('user', None)
+            secret = item.get('secret', False)
+            ttl = item.get('ttl', None)
+
+            # if the value is not a string, convert it to JSON
+            # all keys in the datastore must strings
+            if not isinstance(value, basestring):
+                value = json.dumps(value)
```

Comment:
Perhaps we should do throw instead of doing this "magical" serialization? I think that would be better.

Reply:
So, i did this on purpose so we can potentially have the following data in the file:

``` yaml
- name: foo.list
  value:
    - a
    - b
    - c
- name: foo.dict
  value:
    a: 1
    b: 2
    c: 3
```

Without the conversion then the user would need to serialize their data into JSON in the file (ugly)

``` yaml
- name: foo.list
  value: '["a", "b", "c"]'
- name: foo.dict
  value: '{"a": 1, "b": 2, "c": 3}'
```

Thoughts?

Reply:
For now I would prefer to throw because we don't have any "explicit" contract on serialization / de-serialization format and it could just cause confusion.

Aka we would just serialize it on the way in and not the way out so I would prefer to keep it simple and explicit.

But maybe if we some how explicitly document that and make it very obvious, I would be fine with that change.

Reply:
So maybe would be the best to change this to throw now or perhaps add a log statement on what is going to happen when such value is encountered.

So this way this PR can be finalized and merged :)